crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -111,6 +111,15 @@ impl<N: NodePrimitives> StaticFileProviderBuilder<N> {
         StaticFileProviderInner::new(path, StaticFileAccess::RO).map(|inner| Self { inner })
     }
 
+    /// Set custom blocks per file for specific segments.
+    pub fn with_blocks_per_file_for_segments(
```

Comment:
can we add more docs here what this does?

especially about what happens when this is changed for an existing node